lib/src/multires_stochastic_texture_synthesis.rs

```diff
@@ -762,6 +704,30 @@ impl Generator {
         let mut total_processed_pixels = 0;
         let max_workers = params.max_thread_count;
 
+        {
+            // now that we have all of the parameters we can setup our initial tree grid
+            self.tree_grid = TreeGrid::new(
+                (self.output_size.width as f32 * 1.1) as u32 + 1,
+                1,
+                (self.output_size.width as f32 * 0.05) as u32 + 1,
+            );
+            // if we already have resolved pixels from an inpaint or multiexample add them to this tree grid
+            let resolved_queue = &mut self.resolved.write().unwrap();
+            let pixels_to_update: Vec<([i32; 2], CoordFlat, Score)> = resolved_queue
+                .drain(..)
+                .map(|a| {
+                    let coord_2d = a.0.to_2d(self.output_size);
+                    ([coord_2d.x as i32, coord_2d.y as i32], a.0, a.1)
+                })
+                .collect();
+            self.flush_resolved(
```

Comment:
Note this changes behavior for inpainting + tiling. Now pixels which are not part of the inpaint region will be taken into account when tiling. I could change this back to the old behavior (where those pixels were ignored) by imho this behavior seem correct

Reply:
Perhaps @anopara could have a look. I'm not quite sure what the implications are with the previous and the new behaviors.

Reply:
Hi, @Mr4k. First of all, this is some awesome work, thank you! Could you please elaborate on "Now pixels which are not part of the inpaint region will be taken into account when tiling" 😅 I'm not quite sure how exactly they will be used during tiling?

Reply:
Hi @anopara. Thanks for making this awesome project! Here is an example of what I'm talking about. Consider the output of the following command.
```
cargo run --release -- --out-size 304x308 --inpaint circles_mask.png --tiling -o imgs/01.png generate circles.png
```
using the following images:
circles.png:
<img width="152" alt="circles" src="https://user-images.githubusercontent.com/1163879/72038290-c52ea980-3255-11ea-9199-96909ef6752a.png">
circles_mask.png:
<img width="152" alt="circles_mask" src="https://user-images.githubusercontent.com/1163879/72038308-cfe93e80-3255-11ea-8b65-3cc1a83b1101.png">

On master this generate this following image:
![without_tiling_inpaint](https://user-images.githubusercontent.com/1163879/72038356-fc9d5600-3255-11ea-8704-4b7e6258d041.png)

On my branch this will generate this image:
![with_tiling_inpaint](https://user-images.githubusercontent.com/1163879/72038383-10e15300-3256-11ea-9f83-b1cecce39825.png)

What I mean is that in my image the inpaint takes into account the pixels on the other side of the image to make it tile. As far as I understand this is because in master when you add the static pixels which are not part of the inpaint region to the rtree (https://github.com/EmbarkStudios/texture-synthesis/blob/897d10aaf41efde9acf4122c7c6d19629a0425d6/lib/src/multires_stochastic_texture_synthesis.rs#L265) you do not "reflect" them across image borders like you do when you add pixels during synthesis (https://github.com/EmbarkStudios/texture-synthesis/blob/897d10aaf41efde9acf4122c7c6d19629a0425d6/lib/src/multires_stochastic_texture_synthesis.rs#L295). What I've done is just to use the same function for both which provides mirroring as a side effect.

Reply:
Ah! That is a very good observation and that is indeed correct. Thank you for spotting this case. I guess, we never tested with/thought of tiling to be generated only on one border 😅 Your proposed change is definitely a more desirable behavior! 👍